### PR TITLE
22830 - Reverse Payment Email Notification

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:*
https://github.com/bcgov/entity/issues/22830

*Description of changes:*
- client email for reverse eft payment action

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/auth.py [hunk 1]
```diff
@@ -106,3 +106,13 @@ def check_auth(business_identifier: str, account_id: str = None, corp_type_code:
             auth_response = {'account': {'id': user.user_name,
                                          'paymentInfo': {'methodOfPayment': PaymentMethod.DIRECT_PAY.value}}}
     return auth_response
+
+
+@user_context
+def get_account_admin_users(auth_account_id, **kwargs):
```
Comment (ochiu) [inline pay-api/src/pay_api/services/auth.py:112]: Moved this from payment_account service, so we can re-use this easily.

**Review by Jxio — APPROVED** [review]

File: pay-api/src/pay_api/services/eft_short_names.py [hunk 2]
```diff
@@ -334,8 +335,40 @@ def _reverse_payment_action(cls, short_name_id: int, statement_id: int):
...
+            'accountName': payment_account.name,
+            'reversedAmount': f'{reversed_amount:,.2f}',
+            'outstandingBalance': f'{outstanding_balance:,.2f}',
+            'statementMonth': statement.from_date.strftime('%B'),
+            'statementNumber': statement.id,
+            'dueDate': datetime.fromisoformat(due_date).strftime('%B %e, %Y')
+        }
+
+        org_admins_response = get_account_admin_users(payment_account.auth_account_id)
+        admins = org_admins_response.get('members') if org_admins_response.get('members', None) else []
```
Comment (seeker25) [inline pay-api/src/pay_api/services/eft_short_names.py:360]: `org_admins_response.get('members', [])` would that work?

File: pay-api/src/pay_api/services/eft_short_names.py [hunk 3]
```diff
@@ -334,8 +335,40 @@ def _reverse_payment_action(cls, short_name_id: int, statement_id: int):
...
         ).flush()
 
+        cls._send_reversed_payment_notification(statement, reversed_credits)
         current_app.logger.debug('>reverse_payment_action')
 
+    @classmethod
+    @user_context
+    def _send_reversed_payment_notification(cls, statement: StatementModel, reversed_amount, **kwargs):
+        payment_account = PaymentAccountModel.find_by_id(statement.payment_account_id)
+        summary_dict: dict = StatementService.get_summary(payment_account.auth_account_id)
```
Comment (seeker25) [inline pay-api/src/pay_api/services/eft_short_names.py:345]: type hint for get_summary?

File: pay-api/src/pay_api/services/email_service.py [hunk 4]
```diff
@@ -71,3 +72,17 @@ def _render_shortname_details_body(shortname: str, amount: str, comment: str, sh
         'url': url
     }
     return template.render(params)
+
+
+def _render_payment_reversed_template(params: Dict) -> str:
```
Comment (seeker25) [inline pay-api/src/pay_api/services/email_service.py:77]: good thank you for using this haha

File: pay-api/src/pay_api/services/eft_short_names.py (hunk 2 above)
Comment (seeker25) [reply pay-api/src/pay_api/services/eft_short_names.py:360]: `org_admins_response.get('members', None) or []`

**Pull request merged**